src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,128 @@
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import static frc.robot.Constants.LimelightConstants.*; 
+
+public class Limelight extends SubsystemBase {
+  /**
+   * Creates a new Limelight.
+   */
+
+  // Create variables for the different values given from the limelight
+  private double xOffset; // Positive values mean that target is to the right of the camera; negative values mean target is to the left. Measured in degrees
+  private double yOffset; // Positive values mean that target is above the camera; negative values mean target is below. Measured in degrees
+  private double ta; // Returns a value of the percentage of the image the target takes
+  private double tv; // Sends 1 if a target is detected, 0 if none are present
+
+  private Integer pipeline; // Used to identify which pipline the limelight uses (0-9)
+
+  // Create a network table for the limelight
+  private final NetworkTable m_limelightTable;
+
+  public Limelight() {
+    // Gets the network table for the limelight
+    m_limelightTable = NetworkTableInstance.getDefault().getTable("limelight");
+  
+    // Reset the default settings and pipleines to the Limelight
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("pipeline").setNumber(1);
+  }
+
+  /**
+   * Returns a value of the offset on the x-axis of the camera to the target. Negative values mean the target is to the left of the camera
+   */
+  public double getXOffset() {
+    return xOffset;
+  }
+
+  /**
+   * Returns true if a target is detected
+   */
+  public boolean isTargetDetected() {
+    return (tv > 0.0);
+  }
+
+  /**
+   * Returns true if the target is within a range of the center crosshair of the camera
+   */
+  public boolean isTargetCentered() {
+    return ((xOffset > -1.5) && (xOffset < 1.5) && (xOffset != 0.0));
+  }
+
+  /**
+   *  Calculates the total angle by adding the mounting angle with the y-axis offset angle of the limelight
+   */
+  public double limelightAngle() {
+    return (kLimelightAngle + yOffset);
+  }
+
+  /**
+   * Return the distance from the limelight to the target (floor distance)
+   */
+  public double limelightDistance() {
+    return (kPortHeight - kLimelightHeight) / Math.cos(Math.toRadians(kLimelightAngle + yOffset) + kLimelightOffset);
+  }
+
+  /**
+   * Chooses which pipeline to use on the limelight and prevents invalid values from being sent
+   * 
+   * @param Pipeline Which pipeline to use on the limelight (0-9)
+   */
+  public void setPipeline(Integer pipeline) {
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("pipeline").setNumber(pipeline);
```

Comment:
This should come after your if-statements, since you don't want to set to a pipeline that doesn't exist.